pkg/controller/subscription/subscription_controller.go

```diff
@@ -146,29 +163,68 @@ func (r *ReconcileSubscription) HandleUpgrades(rhmiSubscription *operatorsv1alph
 			Namespace: r.operatorNamespace,
 		},
 	}
-	err = r.client.Get(r.context, k8sclient.ObjectKey{Name: config.Name, Namespace: config.Namespace}, config)
+	err = r.client.Get(ctx, k8sclient.ObjectKey{Name: config.Name, Namespace: config.Namespace}, config)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	err = rhmiConfigs.UpdateStatus(r.context, r.client, config, latestRHMIInstallPlan)
+	objectKey := k8sclient.ObjectKey{
+		Name:      rhmiSubscription.Spec.CatalogSource,
+		Namespace: rhmiSubscription.Spec.CatalogSourceNamespace,
+	}
+	csvFromCatalogSource, err := r.catalogSourceClient.GetLatestCSV(objectKey, rhmiSubscription.Spec.Package, rhmiSubscription.Spec.Channel)
 	if err != nil {
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("error getting the csv from catalogsource %w", err)
 	}
 
-	canUpgradeNow, err := rhmiConfigs.CanUpgradeNow(config, installation)
-	if err != nil {
-		return reconcile.Result{}, err
+	currentOperatorVersionName := fmt.Sprintf("%s.v%s", CSVNamePrefix, version.Version)
+	if csvFromCatalogSource.Spec.Replaces != currentOperatorVersionName {
+
+		if csvFromCatalogSource.Spec.Replaces != latestRHMICSV.Spec.Replaces {
```

Comment:
What if an instance is at 2.3.0, then we release 2.4.0 (replaces 2.3.0) but before the upgrade happens we release 2.3.1. (also replaces 2.3.0) => this condition would resolve to false and IP wouldn't be deleted?

Reply:
in this situation I'd expect the 2.4.0 to change replaces field to 2.3.1?

Reply:
In that case even if we release 2.3.1, the `csvFromCatalogSource` is still 2.4.0, only that now the `replaces` has been updated to 2.3.1, so in that case both conditions would be true, as the current available CSV (latestRHMICSV) is also 2.4.0 but still replaces 2.3.0

Reply:
but if latestRHMICSV replaces != csvFromCatalogSource replaces, then delete the install plan, right?

Reply:
@philbrookes yes, we would need to push another csv for 2.4.0 with the replaces field pointing to 2.3.1

Reply:
Ok, thanks for explanation guys.
I didn't know we would change 2.4.0 replaces field.